mlir/lib/Conversion/PresburgerToStandard/PresburgerToStandard.cpp

```diff
@@ -0,0 +1,130 @@
+    }
+    return condition;
+  }
+
+  Value lowerFlatAffineConstraints(const FlatAffineConstraints &basicSet) {
+    if (basicSet.isEmpty()) {
+      return builder.create<ConstantIntOp>(loc, 0, 1);
+    }
+
+    Value condition = builder.create<ConstantIntOp>(loc, 1, 1);
+    for (unsigned i = 0, e = basicSet.getNumEqualities(); i < e; ++i) {
+      ArrayRef<int64_t> eq = basicSet.getEquality(i);
+      Value isConsSat = lowerPresburgerConstraint(eq.drop_back(), eq.back(),
+                                                  CmpIPredicate::eq);
+      condition = builder.create<AndOp>(loc, condition, isConsSat);
+    }
+
+    for (unsigned i = 0, e = basicSet.getNumInequalities(); i < e; ++i) {
+      ArrayRef<int64_t> ineq = basicSet.getInequality(i);
+      Value isConsSat = lowerPresburgerConstraint(ineq.drop_back(), ineq.back(),
+                                                  CmpIPredicate::sge);
+      condition = builder.create<AndOp>(loc, condition, isConsSat);
+    }
+    return condition;
+  }
+
+  Value lowerPresburgerConstraint(ArrayRef<int64_t> coeffs, int64_t c,
+                                  CmpIPredicate pred) {
+    assert(coeffs.size() == dimAndSymbolValues.size() &&
+           "expect coefficients for every dim and symbol");
+
+    Value sum = builder.create<ConstantIndexOp>(loc, c);
+
+    // TODO can we do a kind of zip().fold() ?
+    for (unsigned i = 0, e = coeffs.size(); i < e; ++i) {
+      Value coeff = builder.create<ConstantIndexOp>(loc, coeffs[i]);
+      Value val = dimAndSymbolValues[i];
+
+      Value prod = builder.create<MulIOp>(loc, coeff, val);
+      sum = builder.create<AddIOp>(loc, sum, prod);
+    }
+
+    Value zeroConstant = builder.create<ConstantIndexOp>(loc, 0);
+    Value cmp = builder.create<CmpIOp>(loc, pred, sum, zeroConstant);
+    return cmp;
+  }
+
+  OpBuilder &builder;
+  ValueRange dimAndSymbolValues;
+  Location loc;
+};
+
+} // namespace
+
+/// Presburger contains are replaced by runtime checks
+class PresburgerContainsLowering : public OpRewritePattern<ContainsOp> {
+public:
+  using OpRewritePattern<ContainsOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(ContainsOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    // TODO find a nicer way to solve this problem
+    // PresburgerSetType setType = op.set().getType().cast<PresburgerSetType>();
+    SetOp setDef = op.set().getDefiningOp<SetOp>();
+    if (!setDef)
+      return failure();
+    PresburgerSetAttr setAttr = setDef.getAttrOfType<PresburgerSetAttr>("set");
+    PresburgerSet set(setAttr.getValue());
+
+    PresburgerSetTransformer t(rewriter, op.dimAndSyms(), loc);
+
+    Value reduced = t.lowerPresburgerSet(set);
+    rewriter.replaceOp(op, reduced);
+    return success();
+  }
+};
+
+void mlir::populatePresburgerToStdConversionPatterns(
+    OwningRewritePatternList &patterns, MLIRContext *ctx) {
+  // clang-format off
```

Comment:
This was done in at most other points where patterns get inserted. If there are multiple patterns, they can be written under each other, but clang-format squashes them in separate lines. 

This will be relevant as soon ad the pwExpr lowering is in place.